lib/puppet/functions/is_absolute_path.rb

```diff
@@ -0,0 +1,10 @@
+Puppet::Functions.create_function(:is_absolute_path, Puppet::Functions::InternalFunction) do
+  dispatch :deprecation_gen do
+    scope_param
+    optional_repeated_param 'Any', :args
```

Comment:
@alexjfisher @DavidS has found the solution this morning and is working on a workaround. And yep it's due to the optional_repeated_param not being available before puppet 3.8.1.